packages/freezed/lib/src/templates/abstract_template.dart

```diff
@@ -51,7 +53,13 @@ ${copyWith.commonContreteImpl(abstractProperties)}
 
   String get _toJson {
     if (!shouldGenerateJson) return '';
-    return 'Map<String, dynamic> toJson() => throw $privConstUsedErrorVarName;';
+    final genericToJsonArgs = hasGenericArgumentFactories
+        ? genericsParameter.typeParameters.map((type) {
+            return 'Object Function($type value) toJson$type';
```

Comment:
```suggestion
            return 'Object? Function($type value) toJson$type';
```

We also need a test to check that toJson can return null

Reply:
`JsonSerializable` does not currently support nullable types, for instance if we accept a nullable type for the `toJson` functions, it complains:
![imagen](https://user-images.githubusercontent.com/32204060/110011616-a6a72f80-7ced-11eb-8611-649321c75f8c.png)

Reply:
Hum that's problematic